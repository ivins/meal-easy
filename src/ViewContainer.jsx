import React from 'react';
import { TabContent, Card, Button, CardText, Col } from 'reactstrap';

export default class ViewContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    const Dietary = () => {
      let dietType = [];
      if (this.props.vegetarian) {
        dietType.push('Vegetarian');
      } 
      if (this.props.vegan) {
        dietType.push('Vegan');
      }
      if (this.props.glutenFree) {
        dietType.push('Gluten-Free');
      }
      if (this.props.dairyFree) {
        dietType.push('Dairy-Free');
      }
      if (dietType.length === 0) {
        return false
      } else {
        return <h6><b>Dietary:</b> {dietType.join(', ')}</h6>;
      }
    }

    const ingredients = this.props.extendedIngredients? this.props.extendedIngredients.map((item)=>{
      return <li className="ingredient-item" key={item.id}><img src={item.image} alt={item.name}/><span>{item.originalString}</span></li>
    }): false;

    return (
      <div id="recipe" className="columnContainer">
        <TabContent >
          <Col sm="11" className="recipeCard">
            <Card body>
              <h3>{this.props.title}</h3><br/>
              <img src={this.props.image} alt={this.props.image}/><br/>
              <h6><b>Servings:</b> {this.props.servings}   <b>Prep:</b> {this.props.preparationMinutes} minutes   <b>Cook:</b> {this.props.cookingMinutes} minutes</h6>
              <Dietary/>
              <br/>
              <h5>Ingredients</h5>
              <ul className="list-group">
                {ingredients? ingredients: <li>Sorry something went wrong! No ingredients were provided.</li>}
              </ul>
              <br/>
              <h5>Instructions</h5>
              <CardText>{this.props.instructions}</CardText><br/>
              <Button>Add to Favourites</Button>
              {this.props.id}
            </Card>
          </Col>
        </TabContent>
      </div>
    );
  }
}