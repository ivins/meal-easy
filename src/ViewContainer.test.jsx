import React from 'react';
import ReactDOM from 'react-dom';
import ViewContainer from './ViewContainer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ViewContainer />, div);
  ReactDOM.unmountComponentAtNode(div);
});
