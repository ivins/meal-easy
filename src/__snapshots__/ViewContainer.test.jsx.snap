// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`construction of View Container with no props renders ViewContainer Component 1`] = `
<div
  className="columnContainer"
  id="recipe"
>
  <TabContent
    tag="div"
  >
    <Col
      className="recipeCardContainer"
      sm="11"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Card
        body={true}
        id="recipeCard"
        tag="div"
      >
        <h3>
          Recipe Panel
        </h3>
        <br />
        <h5>
          Select a recipe to view
        </h5>
      </Card>
    </Col>
  </TabContent>
</div>
`;

exports[`construction of View Container with recipe props renders ViewContainer Component 1`] = `
<div
  className="columnContainer"
  id="recipe"
>
  <TabContent
    tag="div"
  >
    <Col
      className="recipeCardContainer"
      sm="11"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Card
        body={true}
        id="recipeCard"
        tag="div"
      >
        <h3>
          Food
        </h3>
        <br />
        <img
          alt="http..."
          src="http..."
        />
        <br />
        <h6>
          <b>
            Servings:
          </b>
           
          4
             
          <b>
            Prep:
          </b>
           
           minutes   
          <b>
            Cook:
          </b>
           
          5
           minutes
        </h6>
        <Dietary />
        <br />
        <h5>
          Ingredients
        </h5>
        <ul
          className="list-group"
        >
          <li
            className="ingredient-item"
            key="7"
          >
            <img
              src="https://spoonacular.com/cdn/ingredients_100x100/gg"
            />
            <span />
          </li>
        </ul>
        <br />
        <h5>
          Instructions
        </h5>
        <CardText
          tag="p"
        >
          food
        </CardText>
        <br />
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
        >
          Add to Favourites
        </Button>
      </Card>
    </Col>
  </TabContent>
</div>
`;
